ext/stackprof/stackprof.c

```diff
@@ -428,24 +464,53 @@ stackprof_record_sample_for_stack(int num)
 
 	prev_frame = frame;
     }
+
+    if (_stackprof.raw) {
+	gettimeofday(&_stackprof.last_sample_at, NULL);
+    }
 }
 
 void
 stackprof_record_sample()
 {
+    int timestamp_delta = 0;
+    if (_stackprof.raw) {
+	struct timeval t;
+	gettimeofday(&t, NULL);
```

Comment:
I looked at `Process.clock_gettime`, and it looks like it only has support for OS X >= 10.12 (and much older versions of linux) https://github.com/ruby/ruby/blob/fe3decb2017af2092d4b2181e27e4cd8f6bbf4fa/process.c#L7129

If that's fine with you, then I'm happy to update this PR to use that instead, and have these stored as doubles rather than as ints.

Reply:
I don't have a strong preference. This is probably fine.

Reply:
Okay, in that case this should be ready to review!